### ISSUE TITLE
Security concerns with the GH_TOKEN

### ISSUE THREAD

**Issue by namdao2000** [electron-userland/electron-builder#5688]
In order to deploy the application and release it on GitHub, my GH_TOKEN requires `package:write` scope on, which also gives a person the permission to delete/modify/add files to the private repo. On the other hand, the only privilege I need in the GH_TOKEN for distribution is for `package:read`, which only allows the person to download the packages. Any chance I can use the `package:write` GH_TOKEN for deployment and `package:read` token for the distributed app? 

Thank you all in advance :D

**Comment by mmaietta** [comment]
That's a great concern.

Looks like it is not possible to use separate tokens at this time.
https://github.com/electron-userland/electron-builder/blob/master/packages/electron-publish/src/gitHubPublisher.ts#L45
```
token = process.env.GH_TOKEN || process.env.GITHUB_TOKEN
```

https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/src/providerFactory.ts#L23-L28
```
const token = (githubOptions.private ? process.env.GH_TOKEN || process.env.GITHUB_TOKEN : null) || githubOptions.token
```

I'd be happy to review a PR if you're interested in discussing a solution.
Contributing doc: https://github.com/electron-userland/electron-builder/blob/master/CONTRIBUTING.md#to-setup-a-local-dev-environment

**Comment by namdao2000** [comment]
Thank you for your reply mmaietta,

I am not good enough to contribute to the project, but I do hope someone out there can see this and fix this issue.

**Comment by jaydoherty** [comment]
I agree with this idea _in theory_. In my opinion, something like this should work:
1. Set your GH_TOKEN or GITHUB_TOKEN environment variable to your "private" token with publishing privileges. 
2. Set "build": {"publish":{"token": "put your public lesser-privileged readonly token here"}}.

Sadly this doesn't work right now because electron-builder will completely ignore GH_TOKEN if "token" is specified directly in the configuration files. I know this from trying it out on electron-builder 22.14.5.

However...
Does GitHub allow tokens with only readonly scope? I thought GitHub only allowed users to create tokens with a minimum of **BOTH** pull **AND** push permission. I figured this was why electron-builder had never bothered to support separate tokens. At least, there is a long discussion on the topic here (with no progress as of December 2021): https://github.community/t/can-i-give-read-only-access-to-a-private-repo-from-a-developer-account/441

**TL;DR:** this feature might be impossible until GitHub allows more granularity in token permissions.

**Comment by dragonDScript** [comment]
I'd rather not use a private repo and that's it. A GH token is very sensible data.
